ui/imports/shared/popups/ContactVerificationRequestPopup.qml

```diff
@@ -14,18 +14,52 @@ import shared.views.chat 1.0
 StatusModal {
     id: root
 
-    property string senderPublicKey: ""
-    property string senderDisplayName: ""
-    property string senderIcon: ""
-    property string challengeText: ""
-    property string responseText: ""
-    property string messageTimestamp: ""
-    property string responseTimestamp: ""
+    property var contactsStore
+    property string publicKey
 
     signal verificationRefused(string senderPublicKey)
     signal responseSent(string senderPublicKey, string response)
 
-    header.title: qsTr("%1 is asking you to verify your identity").arg(root.senderDisplayName)
+    function updateContactDetails() {
+        try {
+            const request = root.contactsStore.getVerificationDetailsFromAsJson(root.publicKey)
+
+            if (request.requestStatus === Constants.verificationStatus.canceled) {
+                root.close()
+            }
+
+            d.senderPublicKey = request.from,
+            d.senderDisplayName = request.displayName
+            d.senderIcon = request.icon
+            d.challengeText = request.challenge
+            d.responseText = request.response
+            d.messageTimestamp = request.requestedAt
+            d.responseTimestamp = request.repliedAt
+        } catch (e) {
+            console.error("Error getting or parsing verification data", e)
+        }
+    }
+
+    Connections {
+        target: root.contactsStore.receivedContactRequestsModel
```

Comment:
This very ugly workaround i'm not going to merge. @jrainville @caybro i think we should figure out some common approach for data updates that should replace getVerificationDetailsFromAsJson() and getContactDetailsFromJson(). Maybe just regular signal from model?

Reply:
Yeah this seems a bit ugly.

There is always the weird way we used to have which is to have an `observed` QtObject. We still use it in one place: the communities. See `observedItem` in communities/view.nim

It's not the best, but it does make using a single Item from a model work and get updates, since it's a `QtObject`

It doesn't have to be done in the PR, as long as you log it and do it after. The only thing I would suggest to make this a little less ugly is change `onCountChanged` to something more appropriate like `contactRequestChanged`. Bonus if it sends the ID and we only rebuild the contact details when it's the right request.